packages/core/src/modules/credentials/protocol/v2/__tests__/v2.ldproof.credentials.propose-offerED25519.test.ts

```diff
@@ -0,0 +1,540 @@
+        id: expect.any(String),
+        createdAt: expect.any(Date),
+        threadId: expect.any(String),
+        connectionId: expect.any(String),
+        state: CredentialState.CredentialReceived,
+      })
+
+      const credentialMessage = await didCommMessageRepository.findAgentMessage({
+        associatedRecordId: faberCredentialRecord.id,
+        messageClass: V2IssueCredentialMessage,
+      })
+
+      expect(JsonTransformer.toJSON(credentialMessage)).toMatchObject({
+        '@type': 'https://didcomm.org/issue-credential/2.0/issue-credential',
+        '@id': expect.any(String),
+        comment: 'V2 Indy Credential',
+        formats: [
+          {
+            attach_id: expect.any(String),
+            format: 'aries/ld-proof-vc@1.0',
+          },
+        ],
+        'credentials~attach': [
+          {
+            '@id': expect.any(String),
+            'mime-type': 'application/json',
+            data: expect.any(Object),
+            lastmod_time: undefined,
+            byte_count: undefined,
+          },
+        ],
+        '~thread': {
+          thid: expect.any(String),
+          pthid: undefined,
+          sender_order: undefined,
+          received_orders: undefined,
+        },
+        '~please_ack': { on: ['RECEIPT'] },
+        '~service': undefined,
+        '~attach': undefined,
+        '~timing': undefined,
+        '~transport': undefined,
+        '~l10n': undefined,
+      })
+    } else {
+      throw new Error('Missing Connection Id')
+    }
+  })
+
+  test('Multiple Formats: Alice starts with V2 (both ld and indy formats) credential proposal to Faber', async () => {
+    testLogger.test('Alice sends (v2 jsonld) credential proposal to Faber')
+    // set the propose options - using both indy and ld credential formats here
+    const credentialPreview = V2CredentialPreview.fromRecord({
+      name: 'John',
+      age: '99',
+    })
+    const testAttributes = {
+      attributes: credentialPreview.attributes,
+      schemaIssuerDid: 'GMm4vMw8LLrLJjp81kRRLp',
+      schemaName: 'ahoy',
+      schemaVersion: '1.0',
+      schemaId: '1560364003',
+      issuerDid: 'GMm4vMw8LLrLJjp81kRRLp',
+      credentialDefinitionId: 'GMm4vMw8LLrLJjp81kRRLp:3:CL:12:tag',
+      linkedAttachments: [
+        new LinkedAttachment({
+          name: 'profile_picture',
+          attachment: new Attachment({
+            mimeType: 'image/png',
+            data: new AttachmentData({ base64: 'base64encodedpic' }),
+          }),
+        }),
+        new LinkedAttachment({
+          name: 'x-ray',
+          attachment: new Attachment({
+            mimeType: 'image/png',
+            data: new AttachmentData({ base64: 'base64encodedpic' }),
+          }),
+        }),
+      ],
+      payload: {
+        schemaIssuerDid: 'GMm4vMw8LLrLJjp81kRRLp',
+        schemaName: 'ahoy',
+        schemaVersion: '1.0',
+        schemaId: '1560364003',
+        issuerDid: 'GMm4vMw8LLrLJjp81kRRLp',
+      },
```

Comment:
Why do we have the same attributes multiple times (top level and in the payload object)? Can we make sure we're reusing the `schemaIssuerDid` / `schemaName` / `schemaVersion` / `schemaId` / `issuerDid` / `credentialDefinitionId` values?

Reply:
I think we can just remove the payload object and use the top level properties everywhere

Reply:
I think we can just remove the payload object and use the top level properties everywhere